File: Data Structures and Algorithms/1-TwoSum/1-TwoSum.py
# Last updated: 6/4/2026, 6:38:24 PM
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
        for i,n in enumerate(nums):
            diff = target - n
            if diff in hashMap:
                return [hashMap[diff],i]
            hashMap[n] = i
        return

File: Data Structures and Algorithms/1-TwoSum/1-TwoSum.cpp
// Last updated: 6/4/2026, 6:38:19 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mpp;
        for (int i=0; i<nums.size(); i++)
        {   
            int diff = target - nums[i];
            if(mpp.find(diff) != mpp.end())
            {
                return {mpp[diff],i};
            }
            mpp[nums[i]] = i;
        }
        return {-1,-1};
    }
};